Work around for build failures with GCC 6 backported from upstream glib.
See https://bugzilla.gnome.org/761550 for details.

diff -Nrup pkg-config-0.29.1.orig/glib/glib/gdate.c pkg-config-0.29.1/glib/glib/gdate.c
--- pkg-config-0.29.1.orig/glib/glib/gdate.c	2016-01-24 22:51:38.000000000 +0100
+++ pkg-config-0.29.1/glib/glib/gdate.c	2017-02-07 17:34:54.372617599 +0100
@@ -2439,6 +2439,9 @@ win32_strftime_helper (const GDate     *
  *
  * Returns: number of characters written to the buffer, or 0 the buffer was too small
  */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+
 gsize     
 g_date_strftime (gchar       *s, 
                  gsize        slen, 
@@ -2549,3 +2552,5 @@ g_date_strftime (gchar       *s,
   return retval;
 #endif
 }
+
+#pragma GCC diagnostic pop
